glass/grf/_solver.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from transformcl import cltocorr, corrtocl
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from numpy.typing import NDArray
+
+from ._core import Transformation, corr, dcorr, icorr
+
+
+def _relerr(dx: NDArray[Any], x: NDArray[Any]) -> float:
+    """Compute the relative error max(|dx/x|)."""
+    q = np.divide(dx, x, where=(dx != 0), out=np.zeros_like(dx))
+    return np.fabs(q).max()  # type: ignore[no-any-return]
+
+
+def solve(
+    cl: NDArray[Any],
+    t1: Transformation,
+    t2: Transformation | None = None,
+    *,
+    pad: int = 0,
+    initial: NDArray[Any] | None = None,
+    cltol: float = 1e-5,
+    gltol: float = 1e-5,
+    maxiter: int = 20,
+    monopole: float | None = None,
+) -> tuple[NDArray[Any], NDArray[Any], int]:
+    """
+    Solve for a Gaussian angular power spectrum.
+
+    Parameters
+    ----------
+    cl : (n,) array
+    t1, t2 : :class:`Transformation`
```

Comment:
Explicit type information is not required in docstrings + other parameters are undocumented
```suggestion
    cl : (n,) array
    t1, t2 : :class:`Transformation`
```